creator-node/src/apiHelpers.js

```diff
@@ -22,6 +22,46 @@ module.exports.handleResponse = (func) => {
   }
 }
 
+/**
+ * Like `handleResponse` but on an interval sends a heartbeat back through
+ * `res.write` as a piece of the JSON result
+ * @param {() => void} func returns the response JSON
+ */
+module.exports.handleResponseWithHeartbeat = (func) => {
+  return async function (req, res, next) {
+    try {
+      // First declare our content type since we will be sending heartbeats back
+      // in JSON
+      res.set('Content-Type', 'application/json; charset=utf-8')
+
+      // set custom CORS headers that's required if you want to response
+      // headers through axios
+      res.set('Access-Control-Expose-Headers', 'CN-Request-ID')
+
+      // Write a key for the heartbeat
+      res.write('{"_h":"')
+
+      // Fire up an interval that will append a single char to the res
+      const heartbeatInterval = setInterval(() => {
+        if (!res.finished) {
+          res.write('1')
+        }
+      }, 5000)
```

Comment:
I think we're going to run into problems if the request finishes in < 5s, because then we won't append any 1s to the result. I tried JSON.parsing a string with an empty field like that and ran into issues, so we probably want to always have at least a single value here?

Reply:
Not sure I follow, the JSON is

```
{
  "_h": "",
  "data": { }
}
```
in that case

Reply:
Oh yes duh, because empty string is a thing